src/Storages/StorageDistributed.cpp

```diff
@@ -270,6 +270,18 @@ QueryProcessingStage::Enum getQueryProcessingStageImpl(const Context & context,
 /// For destruction of std::unique_ptr of type that is incomplete in class definition.
 StorageDistributed::~StorageDistributed() = default;
 
+static const ColumnsDescription DISTRIBUTED_VIRTUALS = ColumnsDescription(
+    {
+        NameAndTypePair("_table", std::make_shared<DataTypeString>()),
+        NameAndTypePair("_part", std::make_shared<DataTypeString>()),
+        NameAndTypePair("_part_index", std::make_shared<DataTypeUInt64>()),
+        NameAndTypePair("_partition_id", std::make_shared<DataTypeString>()),
+        NameAndTypePair("_sample_factor", std::make_shared<DataTypeFloat64>()),
```

Comment:
Not sure, but I guess the original comment should stay, since all above is not for distributed engine but mostly from mergetree?

```
/// NOTE This is weird. Get rid of this.
```